apps/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,44 @@
 import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
+import {
+  Component,
+  Input,
+  TemplateRef,
+  Output,
+  EventEmitter,
+} from '@angular/core';
 import { StudentStore } from '../../data-access/student.store';
 import { TeacherStore } from '../../data-access/teacher.store';
+import { CityStore } from '../../data-access/city.store';
 import { CardType } from '../../model/card.model';
 import { ListItemComponent } from '../list-item/list-item.component';
+import { CommonModule } from '@angular/common';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, CommonModule],
+  //providers: [TemplateRef],
 })
 export class CardComponent {
   @Input() list: any[] | null = null;
   @Input() type!: CardType;
   @Input() customClass = '';
+  @Input() templateView: TemplateRef<any>;
+  @Output() deleteEvent = new EventEmitter<number>();
+  @Output() addEvent = new EventEmitter();
 
   CardType = CardType;
 
   constructor(
     private teacherStore: TeacherStore,
-    private studentStore: StudentStore
+    private studentStore: StudentStore,
+    private cityStore: CityStore
```

Comment:
This component is a shared component, you should not have any logic inside here. 

use `@Output()` to emit event outside a dumb component, and the logic will be handle by the parent

Reply:
those were unused imports, I missed that and removed them in this commit.